shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -372,6 +436,17 @@ - (void)updateTextAndSelection {
   }
 }
 
+- (NSString*)textAffinityString {
+  return (self.textAffinity == FlutterTextAffinityUpstream) ? kTextAffinityUpstream
+                                                            : kTextAffinityDownstream;
+}
+
+- (NSRange)clampSelection:(NSRange*)range forText:(NSString*)text {
```

Comment:
What about `clampRange:(NSRange*)range forText:(NSString*)text` in case we ever use this on e.g. mark text?

Reply:
Actually it looks like I've removed all usage of this 🤷.  I'll remove it altogether then.